File: sdk/logging/dotnet/PrimusSaaS.Logging/Targets/ConsoleTarget.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using PrimusSaaS.Logging.Core;

namespace PrimusSaaS.Logging.Targets;

/// <summary>
/// Console output target with optional pretty printing
/// </summary>
public class ConsoleTarget : ITarget
{
    private readonly bool _pretty;

    public ConsoleTarget(bool pretty = false)
    {
        _pretty = pretty;
    }

    public void Write(LogEntry logEntry)
    {
        if (_pretty)
        {
            PrettyPrint(logEntry);
        }
        else
        {
            Console.WriteLine(logEntry.ToJson());
        }
    }

    private void PrettyPrint(LogEntry logEntry)
    {
        var color = GetColor(logEntry.Level);
        var timestamp = logEntry.Timestamp.ToLocalTime().ToString("hh:mm:ss tt");
        
        Console.ForegroundColor = ConsoleColor.Gray;
        Console.Write($"[{timestamp}] ");
        
        Console.ForegroundColor = color;
        Console.Write($"{logEntry.Level.ToString().ToUpperInvariant()}: ");
        
        Console.ResetColor();
        Console.WriteLine(logEntry.Message);

        if (logEntry.Context.Count > 0)
        {
            Console.ForegroundColor = ConsoleColor.DarkGray;
            Console.WriteLine(JsonSerializer.Serialize(logEntry.Context, PrettyOptions));
            Console.ResetColor();
        }
    }

    private ConsoleColor GetColor(LogLevel level)
    {
        return level switch
        {
            LogLevel.Debug => ConsoleColor.Cyan,
            LogLevel.Info => ConsoleColor.Green,
            LogLevel.Warning => ConsoleColor.Yellow,
            LogLevel.Error => ConsoleColor.Red,
            LogLevel.Critical => ConsoleColor.Magenta,
            _ => ConsoleColor.White
        };
    }

    public void Close()
    {
        // Nothing to close for console
    }

    private static readonly JsonSerializerOptions PrettyOptions = new()
    {
        WriteIndented = true,
        DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
        ReferenceHandler = ReferenceHandler.IgnoreCycles
    };
}

File: sdk/logging/nodejs/src/targets/ConsoleTarget.ts
import { Target } from './Target';
import { LogEntry } from '../core/LogEntry';
import { LogLevel } from '../core/LogLevel';

/**
 * Target that writes logs to the console
 * Supports pretty printing for development
 */
export class ConsoleTarget implements Target {
    constructor(private options: { pretty?: boolean } = {}) { }

    write(logEntry: LogEntry): void {
        if (this.options.pretty) {
            this.prettyPrint(logEntry);
        } else {
            console.log(JSON.stringify(logEntry));
        }
    }

    private prettyPrint(entry: LogEntry): void {
        const color = this.getColor(entry.level);
        const reset = '\x1b[0m';
        const gray = '\x1b[90m';

        const timestamp = new Date(entry.timestamp).toLocaleTimeString();
        const contextStr = Object.keys(entry.context).length > 0
            ? `\n${gray}${JSON.stringify(entry.context, null, 2)}${reset}`
            : '';

        console.log(
            `${gray}[${timestamp}]${reset} ${color}${entry.level}${reset}: ${entry.message}${contextStr}`
        );
    }

    private getColor(level: LogLevel): string {
        switch (level) {
            case LogLevel.DEBUG: return '\x1b[36m'; // Cyan
            case LogLevel.INFO: return '\x1b[32m'; // Green
            case LogLevel.WARNING: return '\x1b[33m'; // Yellow
            case LogLevel.ERROR: return '\x1b[31m'; // Red
            case LogLevel.CRITICAL: return '\x1b[41m\x1b[37m'; // White on Red
            default: return '\x1b[37m'; // White
        }
    }
}
